src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector.kt

```diff
@@ -113,7 +117,16 @@ object ImportCandidatesCollector {
         val traitsDataLazy = lazy(LazyThreadSafetyMode.NONE) {
             val importContext = ImportContext.from(context, ImportContext.Type.OTHER) ?: return@lazy null
             val modPaths = importContext.getAllModPaths()
-            val traitPaths = modPaths.flatMapTo(hashSetOf()) { importContext.getAllTraitsPathsInMod(it) }
+            val excludedPaths = importContext.project.getExcludedPaths(usedForMethod = true)
+            val traitPaths = modPaths.flatMapTo(hashSetOf()) {
+                importContext.getAllTraitsPathsInMod(it, excludedPaths)
+            }
+            // Suggest methods from excluded trait which is already imported
+            traitPaths += importContext.rootModData.visibleItems.values.flatMap { perNs ->
```

Comment:
What about local imports?

Reply:
Discussed in private to fix this separately